Source/CarthageKit/Project.swift

```diff
@@ -884,17 +884,20 @@ public func cloneOrFetchProject(project: ProjectIdentifier, preferHTTPS: Bool, c
 							return SignalProducer(value: (.Fetching(project), repositoryURL))
 								.concat(fetchRepository(repositoryURL, remoteURL: remoteURL, refspec: "+refs/heads/*:refs/heads/*").then(.empty))
 						}
+
 						// If we've already cloned the repo, check for the revision, possibly skipping an unnecessary fetch
 						if let commitish = commitish {
-							return commitExistsInRepository(repositoryURL, revision: commitish)
+							return referenceExistsInRepository(repositoryURL, pattern: commitish)
+								.zipWith(commitExistsInRepository(repositoryURL, revision: commitish))
```

Comment:
I'd prefer to use the `zip` function here instead. I think that reads more clearly. But I'll accept this version if you prefer it.

```swift
return zip(
        referenceExistsInRepository(repositoryURL, pattern: commits),
        commitExistsInRepository(repositoryURL, revision: commitish)
    )
```